specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,741 @@
+
+ “poolId”: “string”
+
+ “snapshotId”: “string”
+
+ “exportLocations” : “[]string”
+```
+##### 4. PUT /v1beta/{tenantId}/file/shares/{shareId}
+* Updates a share.
+##### Request
+```json
+  “name” : “string”
+
+  “description” : “string”
+```  
+ ##### Response parameters
+ ```json
+ “id” : “string”
+
+ “protocols” : “[]string”
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “tenantId” : “string”
+
+ “userId” : “string“
+
+ “name” : “string”
+
+ “description” : “string”
+
+ “size” : “int64”
+
+ “availabilityZone”: “string”
+
+ “status”: “string”
+
+ “profileId”: “string”
+
+ “poolId”: “string”
+
+ “snapshotId”: “string”
+```
+##### 5. 	DELETE /v1beta/{tenantId}/file/shares/{shareId}
+* Deletes a share
+##### Request
+```json
+    “id” : “string”
+```
+##### 6.	GET /v1beta/{tenantId}/file/snapshots
+* Lists all share snapshots
+#### Request
+```json
+  “tenantId”: “string”
+```
+ ##### Response parameters
+```json
+ “id”: “string”
+
+ “protocols” : "[]string"
+
+ “createdAt” : “2019-03-20T12:49:00.497Z”
+
+ “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+ “name” : “string”
+
+ “description” : “string”
+ 
+ "shareId" : "string"
+
+ “shareSize” : ”int64”
+
+ “snapshotSize” : “int64”
+
+ “status”: “string”
+```
+##### 7.	GET /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Show details of snapshot
+##### Request
+```json
+  “id”: “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+  
+  “shareId” : “string“
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 8.   POST /v1beta/{tenantId}/file/snapshot
+* Creates a snapshot from a share.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+
+  “shareId” : “string“
+```
+  ##### Response parameters
+```json  
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 9.	PUT /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Updates a share snapshot.
+##### Request
+```json
+  “description” : “string”
+
+  “name” : “string”
+```  
+##### Response parameters
+```json
+  “id”: “string”
+
+  “shareId” : “string”
+
+  “protocols” : "[]string"
+
+  “createdAt” : “2019-03-20T12:49:00.497Z”
+
+  “updatedAt” : “2019-03-20T12:49:00.497Z”
+
+  “name” : “string”
+
+  “description” : “string”
+
+  “shareSize” : “int64”
+
+  “snapshotSize” : “int64”
+
+  “status”: “string”
+```
+##### 10.	DELETE /v1beta/{tenantId}/file/snapshots/{snapshotId}
+* Deletes a share
+##### Request
+```json
+  “id” : “sting”
+```
+##### 11.	POST /v1beta/{tenantId}/file/accesses/
```

Comment:
API endpoint is confusing, acls are defined for file or fileshare. It can be POST /v1beta/{tenantId}/file/shares/{shareId}acls/

Is it necessary that we should define API for defining acls, Can't we use the PUT fileshare  API to provide acls?

PUT /v1beta/{tenantId}/file/shares/{shareId}
"accesscontrol": {
      “accessCapability” : read/write/execute
       other parameters etc.
}

Reply:
I think endpoint is clear. we have to give shareId as mandatory parameter to provide acls on fileshare.

It is recommended that acl's should not set while creating the file share(It should be synchronous). And that is the reason we have separate API for that.